python/lsst/daf/butler/core/registry.py

```diff
@@ -76,6 +78,185 @@ class RegistryConfig(ConfigSubset):
     defaultConfigFile = "registry.yaml"
 
 
+class ConnectionStringBuilder:
```

Comment:
I tried, but then because I cast whatever `config` is into a `RegistryConfig` I tend to end up with circular import errors. This happens in the new implementation too. I'll work on it tomorrow, and it will get moved out to lets say `connectionString.py`.

Reply:
Fixed. Moved into its own file now.